File: server/php/src/Infrastructure/Mappers/PaymentDetails/OperationPaymentReferencesMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\OperationPaymentReferences as OperationPaymentReferencesDto;
use OnlinePayments\Sdk\Domain\OperationPaymentReferences as OperationPaymentReferencesSdk;

final class OperationPaymentReferencesMapper
{
    public static function mapFromSdkResponse(?OperationPaymentReferencesSdk $response): OperationPaymentReferencesDto
    {
        $dto = new OperationPaymentReferencesDto();
        $dto->merchantReference = $response?->merchantReference ?? null;

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/OperationPaymentReferencesMapper.cs
using OperationPaymentReferencesSdk = OnlinePayments.Sdk.Domain.OperationPaymentReferences;
using OperationPaymentReferencesDto = Business.Domain.Payments.PaymentDetails.OperationPaymentReferences;

namespace Infrastructure.Mappers.PaymentDetails;

public static class OperationPaymentReferencesMapper
{
    public static OperationPaymentReferencesDto Map(OperationPaymentReferencesSdk? response)
    {
        return new()
        {
            MerchantReference = response?.MerchantReference
        };
    }
}